feat: DescriptionLength rule

a-frantz opened the pull request:
_Describe the problem or feature in addition to a link to the issues._

Partially addresses #224 by adding a new lint rule.

Before submitting this PR, please make sure:

For external contributors:

- [ ] You have read the `wdl` crates' [CONTRIBUTING](https://github.com/stjude-rust-labs/wdl/blob/main/CONTRIBUTING.md) guide in its entirety.
- [ ] You have not used AI on any parts of this pull request.

For all contributors:

- [x] You have added a few sentences describing the PR here.
- [x] You have added yourself or the appropriate individual as the assignee.
- [x] You have added at least one relevant code reviewer to the PR.
- [x] Your code builds clean without any errors or warnings.
- [x] You have added tests (when appropriate).
- [x] You have added an entry in the CHANGELOG (when appropriate).
- [x] You have updated the README or other documentation to account for these changes (when appropriate).
- [x] You have either (a) made a PR to the `next` branch in the sprocket.bio repository [if you are a Sprocket team member] or (b) have suggested any changes you _think_ need to be made to sprocket.bio in the description of your PR [if you are an external contributor].

a-frantz commented on crates/wdl-lint/RULES.md:
I noticed some of these tags were incorrect, so I opted to remove that whole column. IMO it's not needed and just add maintenance burden

a-frantz commented on crates/wdl-lint/src/rules/description_length.rs:
```diff
@@ -0,0 +1,203 @@
+        ID
+    }
+
+    fn description(&self) -> &'static str {
+        "Ensures that `description` meta entries are not too long for display in documentation."
+    }
+
+    fn explanation(&self) -> &'static str {
+        "Descriptions should be kept short so that they can always render in full. If a \
+         `description` is too long, it may be clipped in some contexts during documentation. \
+         `help` meta entries are permitted to be of any length, and may be a better place for long \
+         form text."
+    }
```
Should these explicitly mention `sprocket [dev] doc`?

a-frantz commented on crates/wdl-lint/src/tags.rs:
```diff
@@ -33,8 +33,11 @@ pub enum Tag {
     Documentation,
+
+    /// Rules associated with compatibility with documentation rendering.
+    DocRendering,
```
I do not love this new tag, but I do not think that this rule should be part of the default set of rules, which includes the `Documentation` tag. Open to hearing alternatives here

a-frantz commented on crates/wdl-lint/src/rules/description_length.rs:
```diff
@@ -0,0 +1,206 @@
+         entries are never clipped and may be a better place for long form text."
+    }
+
+    fn tags(&self) -> TagSet {
+        TagSet::new(&[Tag::SprocketDoc])
+    }
```
Yea that might be a better option 🤔 Although whatever is decided here will also be used in #353 and #354 , so there would be at least 3 rules with this tag

claymcleod commented on crates/wdl-ast/CHANGELOG.md:
```diff
@@ -7,6 +7,10 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
+#### Added
+
+* Documentation change to specify what happens when `text()` is called on an empty `LiteralString` ([#351](https://github.com/stjude-rust-labs/sprocket/pull/351)).
```
I don't really think this requires a changelog entry to be honest. If you still feel it does, at least adding it as like a `#### Chore` or something.

claymcleod commented on crates/wdl-lint/src/rules/meta_description.rs:
```diff
@@ -58,14 +58,17 @@ impl Rule for MetaDescriptionRule {
-         active voice and complete sentences. More detailed information can be included in the \
-         `help` key."
+        "Each task, workflow, and struct should have a description in the meta section. The \
+         description should be short, written in active voice, and be in complete sentences. More \
```
```suggestion
         description should be short, written in active voice and be in complete sentences. More \
```

claymcleod commented on crates/wdl-lint/RULES.md:
```diff
@@ -1,53 +0,0 @@
-# Rules
```
I don't view this as a good change, as it's the only place where there is a plaintext description of each of the rules that are available, what their tags are, and what a brief description is. I understand that it sometimes gets out of sync, but another solution is just to ensure we have a check to make sure this table is updated in the PR checklist.

For me, I don't think we should delete this until we have a better strategy around publishing the available rules and associated information in the docs.

claymcleod replied:
And I didn't say this explicitly, but I don't think asking non-Rust-familiar users to dig through our developer documentation to find out this information is a good enough solution in the meantime.

a-frantz commented:
@claymcleod I added back `RULES.md`, but have removed the table formatting. IMO it's just an extra maintenance burden that only functions to prettify the source, which is probably not how most people are viewing it. I also added the PR template two lint rule specific checks to make sure that these updates aren't forgotten in the future
